tree/treeplayer/inc/TTreeReader.h

```diff
@@ -264,13 +274,15 @@ class TTreeReader: public TObject {
 
    enum EStatusBits {
       kBitIsChain = BIT(14), ///< our tree is a chain
-      kBitHaveWarnedAboutEntryListAttachedToTTree = BIT(15) ///< the tree had a TEntryList and we have warned about that
+      kBitHaveWarnedAboutEntryListAttachedToTTree = BIT(15), ///< the tree had a TEntryList and we have warned about that
+      kBitSetEntryBaseCallingLoadTree = BIT(16) ///< SetEntryBase is in the process of calling TChain/TTree::LoadTree.
    };
 
    TTree* fTree = nullptr; ///< tree that's read
    TEntryList* fEntryList = nullptr; ///< entry list to be used
    EEntryStatus fEntryStatus = kEntryNotLoaded; ///< status of most recent read request
-   Int_t fMostRecentTreeNumber = -1; ///< TTree::GetTreeNumber() of the most recent tree
+   ELoadTreeStatus fLoadTreeStatus = kLoadTreeNone; ///< Indicator on how LoadTree was called 'last' time.
+   std::unique_ptr<TNotifyLink<TTreeReader>> fNotify; // Callback object used by the TChain to update this proxy
```

Comment:
Can this be `TNotifyLink<TTreeReader> fNotify`?

Reply:
maybe ... we also need a trigger/flag for 'has been added to the TTree' ...

Reply:
Could `fNotify.fCurrent` serve that purpose?